### ISSUE TITLE
how can i increase the accuracy of training in yolov5s model

### ISSUE THREAD

**Issue by SaraMashal** [ultralytics/yolov5#12561]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trained a yolov5s model for trash and non-trash object detection, I have a custom dataset 15k images and 21k annotations divided between 2 classes (trash and non-trash). I trained the dataset on the standard yolov5s model and these are the model outputs :
![Screenshot from 2023-12-31 17-01-13](https://github.com/ultralytics/yolov5/assets/40832926/2343ea3b-2889-483c-9e5f-1031f881f85f)
also here is the performance charts:
![media_images_Results_25_0](https://github.com/ultralytics/yolov5/assets/40832926/83db5757-0766-4793-a498-65c2797ed656)
--------------------
now my question is why the accuracy is low? 
also when testing i have issues of overlapping objects like this :
![test2](https://github.com/ultralytics/yolov5/assets/40832926/bc5cdb09-940a-4e88-8a8e-4c78e312b326)
and even though the accuracy of the non-trash class is less than the trash class, most of the objects get detected as non-trash...
for example if i got an image with all trash items it detects better, like this one:
![test3](https://github.com/ultralytics/yolov5/assets/40832926/2a558163-0760-4b97-a085-4cd92c913ffc)
thank you for help...

**Comment by glenn-jocher** [comment]
@SaraMashal to improve YOLOv5s model accuracy, ensure a balanced dataset, adequate augmentation, optimization of hyperparameters, and training for more epochs. For overlapping objects, consider non-maximum suppression. For class imbalances, try data augmentation or methods like weighted loss functions. Additionally, check if the dataset accurately represents real-world scenarios and consider increasing the dataset size if possible. Happy to help!

**Comment by Hyrtsi** [comment]
I think the answer by glenn-jocher was already quite good. I'm going to add something little just for my own entertainment and the possibility to help you.

How do you define trash? If the coke bottle in your picture, for example, had the bottle cap on would it be trash? If the context was that it's on a table instead of on the floor would it be trash? I think the concept of "trashness" is vague and may lead to problems. You should address this from the start before you start collecting, annotating or cleaning data let alone training any models.

What is your expected accuracy? I mean, your goal? Which accuracy would you be happy with?

Why do you expect `yolov5s` be the right model for this job? Have you considered any other models? Is the mere bounding box + class enough for your usecase or should you consider segmentation, for instance?

Is your model and dataset somewhere publicly available so we could take a look?

> when testing i have issues of overlapping objects

How are they annotated in your dataset? Without seeing your dataset I would guess that the piece of trash is a false negative instead of the labels being wrong. You need more samples in the dataset that include this kind of scenario (overlapping bounding boxes of obstacles and trash). You can use your favorite data visualization tools to check the distributions of your labels in the dataset to see if there are some scenarios missing. The `yolov5` repository already has some capabilities for this. Consider checking those.

>  and even though the accuracy of the non-trash class is less than the trash class, most of the objects get detected as non-trash...

What do you mean with non-trash class? The obstacles? This sounds like your dataset is skewed towards non-trash. Let's take a little example.

Imagine that you have a cat detector. It detects cats from images. It gives just a probability if there's a cat in the picture. If you train such model with 1000 images, 900 of which have cats and test it with similar dataset, a model that is simply `return 1.0` will get an accuracy of 90%, right? 

Similarly, if your dataset has 90% of the time non-trash, your model will perform better if it just guesses that the class of the object is non-trash. I'm simplifying this idea overly much but I hope you get my point. Your dataset needs to be balanced like glenn-jocher said in his answer. Otherwise you will face such problems.

I will close with this thought: machine learning is 99% about data. Ensure that your data is of really high quality. Browse through a sample of images with annotations from each class (training, test*). If you think your data is of high enough quality then your problem is that there's not enough of it or it's imbalanced or your model is the problem. However, I believe that data is the problem 99% of the times.

*the dataset that you use to validate whether you accept the model or not should remain secret to keep the process unbiased

Good luck!

**Comment by SaraMashal** [comment]
thank you so much guys, your replies really helped us know where to start from...
and thats true I have figured that the issue was purely from the data annotations, they were not the best, some were missing and some were poorly annotated, and some images were so biased to one environmental setting, we also figured out that the non-trash annotations, were so scattered over the images to the point it selected some useless objects as obstacles, the dataset in the initial place was collected from different places all over the internet, some had ready annotations, the rest were annotated using an AI model for annotating  objects, and we didnt pay much attention to the ready data because we thought its good, for the fact it was used in other models that trained well and gave good results...
Now we refining the data and hoping for better results in the next training sessions!

**Comment by glenn-jocher** [comment]
@SaraMashal That's great to hear! It sounds like you've made some important discoveries regarding your dataset. Quality annotations and a well-curated dataset are indeed crucial for training effective models. Refining your data is a step in the right direction. Remember, consistency in annotations and representing the diversity of scenarios your model will encounter in the real world are key. Best of luck with your next training sessions, and if you have any more questions or need further assistance, feel free to reach out. Keep up the good work! 🚀👍
